cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -403,20 +403,63 @@ def vm(self, rest_api, vm_name):
         assert result.name == vm_name
         return result
 
-    def test_power_off(self, verify_vm_running, vm):
+    def verify_vm_rest_status(self, vm, status):
+        wait_for(lambda: vm.power_state == status,
+            num_sec=300, delay=5, fail_func=vm.reload,
+            message='wait for VM to stop (current state: {})'.format(vm.power_state))
+
+    def check_finish(self, provider, vm, status):
+        logger.info("VM: {} Power State: {} State on provider: {}".format(
+            vm.name, vm.power_state, provider.mgmt.vm_status(vm.name)))
+        return vm.power_state == status
+
+    def test_stop_detail(self, rest_api, provider, verify_vm_running, vm):
         assert "stop" in vm.action
+        self.verify_vm_rest_status(vm, status='on')
         vm.action.stop()
-        wait_for(lambda: vm.power_state == "off", num_sec=300, delay=5, fail_func=vm.reload)
+        wait_for(lambda: self.check_finish(provider, vm, status="off"),
+            num_sec=600, delay=10, fail_func=vm.reload,
+            message='wait for VM to stop (current state: {})'.format(vm.power_state))
 
-    def test_power_on(self, verify_vm_stopped, vm):
+    def test_power_on_detail(self, rest_api, provider, verify_vm_stopped, vm):
         assert "start" in vm.action
+        self.verify_vm_rest_status(vm, status='off')
         vm.action.start()
-        wait_for(lambda: vm.power_state == "on", num_sec=300, delay=5, fail_func=vm.reload)
+        wait_for(lambda: self.check_finish(provider, vm, status="on"),
+            num_sec=600, delay=10, fail_func=vm.reload,
+            message='wait for VM to start (current state: {})'.format(vm.power_state))
 
-    def test_suspend(self, verify_vm_running, vm):
+    def test_suspend_detail(self, rest_api, provider, verify_vm_running, vm):
```

Comment:
The `verify_vm_running` fixture here should make sure that the vm used is... well... running. In case of RHEV (which was causing us problems), it checks using the mgmt_system if the power state is 'up'. CFME translates this into 'on'.
You described the behavior as 'VM being destroyed during provisioning when asked to be stopped (through the REST API)' which leads me to believe that one of the following two things happens:
1. We request a VM, mgmt_systems tells us it is provisioned and 'up', CFME however thinks its still being provisioned, we ask CFME to stop the VM and CFME assumes that since its being provisioned, we are actually asking it to delete it
or (less likely)
2. `verify_vm_running` fixture needs to be fixed/enhanced with regards to RHEV because it is lying to us